lib/parse.js

```diff
@@ -1,54 +1,52 @@
 "use strict";
 
 const babylonToEspree = require("./babylon-to-espree");
-const parse = require("@babel/parser").parse;
-const tt = require("@babel/parser").tokTypes;
-const traverse = require("@babel/traverse").default;
-const codeFrameColumns = require("@babel/code-frame").codeFrameColumns;
+const { parseSync: parse, tokTypes: tt, traverse } = require("@babel/core");
 
-module.exports = function(code, options) {
-  const legacyDecorators =
-    options.ecmaFeatures && options.ecmaFeatures.legacyDecorators;
+/*
+ * A filePath may not exist when the --stdin option is used on the command line or
+ * CLIEngine#executeOnText() is used programmatically. When it doesn't exist, ESLint sets
+ * the filePath to "<text>".
+ *
+ * https://eslint.org/docs/user-guide/command-line-interface#using-stdin
+ * https://eslint.org/docs/developer-guide/nodejs-api#cliengineexecuteontext
+ * https://github.com/eslint/eslint/blob/master/lib/cli-engine.js#L177
+ */
+const ESLINT_DEFAULT_FILEPATH = "<text>";
```

Comment:
I just came across [this](https://github.com/eslint/eslint/blob/master/lib/linter.js#L308) - going to talk to the team and make sure I'm understanding all the possible values we use when there isn't a filename provided. Seems confusing, so just want to make sure I understand the rationale here. Will update when I know more.

Reply:
Given that any path could be provided by a consumer, I think it's best to not try to guess the user's intent here and to just pass through what babel-eslint is given.